### PR TITLE
 Modernized ESProducers in TrackingTools/KalmanUpdators

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Removed headers and put declaration in .cc files
- Removed use of module.cc file
- Pulled out parameters in constructor
- Wrote fillDescriptions
- Used edm::ESGetToken
- Made KFSwitching1DUpdator not depend on ParameterSet

#### PR validation:

The code compiles. This is a refactoring and should not change any results.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: TrackingTools/KalmanUpdators/plugins/TrackingRecHitPropagatorESProducer.cc [hunk 1]
```diff
@@ -1,32 +1,48 @@
...
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/KalmanUpdators/interface/TrackingRecHitPropagator.h"
+#include <memory>
 
 #include "FWCore/Framework/interface/EventSetup.h"
 #include "FWCore/Framework/interface/ESHandle.h"
 #include "FWCore/Framework/interface/ModuleFactory.h"
 #include "FWCore/Framework/interface/ESProducer.h"
```
Comment (perrotta) [inline TrackingTools/KalmanUpdators/plugins/TrackingRecHitPropagatorESProducer.cc]: Repetition of L1

File: TrackingTools/KalmanUpdators/plugins/TrackingRecHitPropagatorESProducer.cc [hunk 2]
```diff
@@ -1,32 +1,48 @@
...
-#include <FWCore/Utilities/interface/ESInputTag.h>
+#include "FWCore/Utilities/interface/ESInputTag.h"
 
 #include <string>
 #include <memory>
 
+class TrackingRecHitPropagatorESProducer : public edm::ESProducer {
+public:
+  TrackingRecHitPropagatorESProducer(const edm::ParameterSet& p);
+  ~TrackingRecHitPropagatorESProducer() override;
```
Comment (perrotta) [inline TrackingTools/KalmanUpdators/plugins/TrackingRecHitPropagatorESProducer.cc]: Should this overridden dtor remain?

File: TrackingTools/KalmanUpdators/plugins/TrackingRecHitPropagatorESProducer.cc (hunk 1 above)
Comment (Dr15Jones) [reply TrackingTools/KalmanUpdators/plugins/TrackingRecHitPropagatorESProducer.cc]: done

**Comment by Dr15Jones** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical changes, in line with the PR description
- Jenkins tests pass and show no difference beside the slimmedJetsAK8 one, which also appears in other unrelated pull requests and it is therefore quite likely a random fluctuation

**Comment by fabiocos** [comment]
+1

**Comment by makortel** [comment]
part of #26748

**Pull request merged**